src/librustc_typeck/check/demand.rs

```diff
@@ -470,7 +474,12 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                         let sugg_expr = if needs_parens { format!("({})", src) } else { src };
 
                         if let Some(sugg) = self.can_use_as_ref(expr) {
-                            return Some(sugg);
+                            return Some((
```

Comment:
I think we're reaching the point where this function should probably return a struct with named fields ;)

Reply:
But I think for consistency that would requires a lot of irrelevant changes. If that's really need would changing them in a seperate PR be better? :-P

Reply:
Oh yea, this was mainly a drive by comment, feel free to address it or not or in a different PR